### PR TITLE
add persist personas endpoint (like mobile)

### PR DESCRIPTION
**Pull request by ahsu1230**
@stkem @yingjieMiao

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala [hunk 1]
```diff
@@ -842,13 +842,17 @@ class LibraryCommander @Inject() (
...
             recipient match {
               case Left(userId) =>
-                libraryMembershipRepo.getWithLibraryIdAndUserId(libraryId, userId) match {
-                  case Some(mem) if mem.access == access =>
-                    None
-                  case _ =>
-                    val newInvite = LibraryInvite(libraryId = libraryId, inviterId = inviterId, userId = Some(userId), access = access, message = msgOpt)
-                    val inviteeInfo = (Left(invitedBasicUsersById(userId)), access)
-                    Some(newInvite, inviteeInfo)
+                if (userId == targetLib.ownerId) {
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala:845]: prevent library invites to that library owner from persisting to database =P

File: kifi-backend/modules/shoebox/conf/shoeboxService.routes [hunk 2]
```diff
@@ -405,6 +405,7 @@ GET     /site/user/settings         @com.keepit.controllers.website.UserControll
 POST    /site/user/settings         @com.keepit.controllers.website.UserController.setSettings()
 
 GET     /site/user/personas             @com.keepit.controllers.website.UserPersonaController.getAllPersonas()
+POST    /site/user/personas              @com.keepit.controllers.website.UserPersonaController.selectPersonas()
 POST    /site/user/personas/:name        @com.keepit.controllers.website.UserPersonaController.addPersona(name: String)
 DELETE  /site/user/personas/:name        @com.keepit.controllers.website.UserPersonaController.removePersona(name: String)
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes:410]: keeping the other endpoints until we're sure whether they'll be used or not

**Comment by ahsu1230** [comment]
the idea of this is on first sight of WAY, we want to persist personas only when "Done" is pressed, so libraries are not created /recos do not appear for unselected personas.

It will take a little longer for recos to appear, but thankfully the FTUE appears after the WAY (for extension users only) so it buys us some time.

**Comment by ahsu1230** [comment]
merging so can work on FE

**Comment by stkem** [comment]
:+1:

**Pull request merged**